sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1079,6 +1079,18 @@ class DataFrameAggregateSuite extends QueryTest
     assert(aggs.head.output.map(_.dataType.simpleString).head ===
       aggs.last.output.map(_.dataType.simpleString).head)
   }
+
+  test("SPARK-33726 Duplicate field name aggregation should not have null values in dataframe") {
+    val query =
+      """|with T as (select id as a, -id as x from range(3)), U as (select id as b,
+         |cast(id as string) as x from range(3)) select T.x, U.x, min(a) as ma, min(b) as mb
+         |from T join U on a=b group by U.x, T.x
+      """.stripMargin
+    val df = spark.sql(query)
+    val nullCount = df.filter($"ma".isNull ).count + df.filter($"mb".isNull ).count
+    + df.filter($"U.x".isNull ).count + df.filter($"T.x".isNull).count
+    assert(nullCount == 0)
```

Comment:
I would suggest to use the `checkAnswer` method for validating the result as it is a consistent set of Rows. You do not need to worry about ordering as the expected and actual data is checked with a set contains.

And a nit: the space before ")" is not needed.

So I suggest this:

```suggestion
    checkAnswer(df, Row(0, "0", 0, 0) :: Row(-1, "1", 1, 1) :: Row(-2, "2", 2, 2) :: Nil)
```

Reply:
Done, thanks for the suggestion.

Reply:
Updated the PR description. Let me know if you have additional suggestions to my description change.